### DISCUSSION TITLE
Non-mandatory flip

### DISCUSSION THREAD

**Discussion by hlolve** [idena-network/idena-docs#181] [IIPs]
Please review and discuss about this

https://github.com/hlolve/non-mandatoryflips/blob/main/README.md

**Comment by sekisanchi** [comment]
Please don't seriously get this comment, but "Whit influx" is not universal, and better to avoid using specific cultural assumptions. Please try using plain and easy description for fundamental mechanism.  Looks similar to my own wish idea but not sure for the referenced previous IIPs. This is a first comment for appreciation, and will be back for the detail.

**Reply by hlolve** [reply]
Proposed changes in rewards affect IIP-7 and is optional to introduce a change in IIP-6. Anyway, the main change is validate in 2 groups, and these changes are only for modify rewards to incentive participants in group#1

**Reply by sekisanchi** [reply]
Splitting human makes people imagine racial/religious conflict, and the expertise on FLIPs does not rely on them. 
So I prefer having expertise/profession like attribute upon current human category, and that's a natural consequence of my experience on this project, and also from protocol design perspectives. 

However, defining characteristic upon the two groups, with or without that attribute is crucial on both protocol design validation and adaptation for the universal public.

**Comment by 0JustinMiles0** [comment]
This proposal significantly reduces sybil resistance properties of Idena. An attacker who would be interested in farming validated identities for a third-party service such as Gitcoin would be able create more validated identities with less humans by just validating profiles in the group 2. If implemented, Idena wouldn't be a reliable sybil-resistant solution for third party apps anymore. 

Moreover, farmers that are interested in farming identities to earn mining and validation rewards could use the group 2 to raise sybil identities to the status of Human in order to acquire full stake protection before moving those to group 1.

Removing IIP-7 seems also a risky choice as it would redirect a significant share of the rewards towards farming pools.

I would add that the motivations of the proposal are unclear. It seems that this proposal assumes that participants from third-party services create less qualitative flips which is unproven and that providing them the option to opt-out from flip making would increase the quality of the flips. Flips quality is highly correlated to the share of farm pools participation hence, by the mechanics mentioned above, I think this proposal would result in a decrease in flips quality.

My first question to author would be clarify what this proposal is attempting to fix/solve.

**Reply by hlolve** [reply]
Maybe we are not seeing the same problem in Idena network. If Idena network wants mass adoption, creating flips is not an easy task and will be more difficult with AI improvements. Passing a validation will be the same for group#1 or group#2 but creating flips should be a more specialized task, and in that case making that mandatory as currently only will benefit AI. Our new adversary is not farm pools, is AI and is closer day by day

**Reply by 0JustinMiles0** [reply]
So would you say that the sybil resistance properties of validated identities in group 1 vs group 2 are equal?

I still don't really understand the motivations for this proposal. Is the main goal of this proposal to make Idena more AI-resistant? Is it to get mass adoption or both?

**Reply by hlolve** [reply]
The proposal is to make Idena easier for not "full" interested people. Runing a node or submit good flips will not prove you are a human, that are current rules. Under these conditions we are affecting really interested participants (bad reporting from users not interested) and ejecting people that just want to prove they are human, and not prove specialized in flip creation against AI

**Reply by 0JustinMiles0** [reply]
Ok, but don't you think that this proposal would reduce the sybil-resistance properties of Idena? 

Also you're saying: "Under these conditions we are affecting *(I think you meant "attracting")* really interested participants". What do you mean by "interested participants"? Do you consider farm pools participating for economic gains "interested participants"?

**Reply by hlolve** [reply]
I am worried that people in group#1 don´t leave the network, that is the heart of Idena, the only that still make the network alive. Group#2 is for mass adoption, and will not affect group#1. You can't expect that each candidate that want to take a test for prove is "human" will be interested in all rules, and if that majority of candidates never learn (not interested) will always affect good and responsible users. Farm pools are not bad if they make good flips. Currently main rewards of farm pools are flips and reports, and also hunting big stake invites

**Reply by 0JustinMiles0** [reply]
You didn't answer my first question so I'll answer it by taking an extreme example in which only 2 identities would create 3 flips each. 

So now imagine that we have 2 identities in Group#1 and 20,000 identities in Group#2 which 50% of them are controlled by one single farmer. Since all identities will see the same 6 flips, this farmer will be able to validate 10,000 identities with just one human and some automation. 

So yes, this proposal would drastically change the sybil resistance properties of Idena.

**Reply by 0JustinMiles0** [reply]
"Farm pools are not bad if they make good flips", yes but history has proven that they do flips of lesser quality than the rest of the network and this proposal isn't fixing this issue while improving the economics for farm pools.

**Reply by hlolve** [reply]
Yes, farm pools still make bad flips and are not reported, that is the main problem

**Reply by hlolve** [reply]
In the proposal is clear that you need a minimum of participants in group#1

**Reply by hlolve** [reply]
I am sure that farm pools will want to go in group#1, but that is limited by minimum stake of their IDs

**Reply by 0JustinMiles0** [reply]
> In the proposal is clear that you need a minimum of participants in group#1

But the proposal doesn't enforce this and it doesn't specify what minimum level is required. This is a critical parameter that needs to be studied.

**Reply by hlolve** [reply]
Yes, I am waiting devs words, this is a draft proposal. In the proposal also says if that number is not reached flips will be mandatory for next epoch as currently. "In that case if at calculating results not many selected the option to be creators, that epoch will be same as current rules, mandatory for all validated IDs"

**Reply by 0JustinMiles0** [reply]
> Yes, farm pools still make bad flips and are not reported, that is the main problem

I don't disagree that this is a problem but it's not a new problem that has emerged with Gitcoin participants. Actually the quality of flips has been improving over time (as tracked by the Flips Quality Survey) which includes the period after the Gitcoin integration was live. I'm not necessarily saying that Gitcoin participants have led to an increase in flips quality but I haven't seen any data to make the case that Gitcoin participants have led to a degraded flips quality.

However, a higher share of independent participants (not controlled by one owner) should lead to higher flips quality because those decrease the share of rewards available for farm pools which degrade their economics. So it is very possible that Gitcoin participants have actually contributed to increase flips quality.

That said, I agree that reporting rules are too complex and not effectively penalizing bad flips. I have suggested participants to adopt a different simplified reporting framework which should lead to a higher share of flips being reported but those guidelines are not official and their impact is likely limited.

**Reply by hlolve** [reply]
If we expect that thousands of new users will learn flips creation rules and apply at time of reporting, Idena network will be a permanent dream. And this is not a new problem

**Reply by hlolve** [reply]
Users that don´t learn the rules get a bad experience for them and for good users at validation ceremony

**Reply by 0JustinMiles0** [reply]
I don't know if there is an easy workaround without any major tradeoff. Currently, a human may validate up to 3 Human profiles but it seems that this proposal would multiply this human to validated identities ratio by a factor X regardless of the chosen parameters for the minimum ratio of identities in group#1 versus group#2. This is not a minor tradeoff and it would drastically change the value proposition of Idena.

**Reply by hlolve** [reply]
With easy template flips not AI resistant and not reported, AI will solve more validations than any real human

**Reply by hlolve** [reply]
We need a minimum quantity of flips for validation, but quality is more important than quantity

**Reply by 0JustinMiles0** [reply]
But the proposal doesn't demonstrate how it would improve the flips quality. You make this reasoning on the assumption that Gitcoin participants don't understand the reporting rules and as such less bad flips are being reported leading to lower flips quality.

However, if you look at the flips reporting stats you'll notice that the ratio of reported flips has remained pretty constant pre and post Gitcoin integration, so the data doesn't support this reasoning.

**Reply by hlolve** [reply]
Is not for new gitcoin hunters, just review Idena history, why network don´t growth ? New gitcoin integration just put this in more evidence. Idena rules are not for an average "not interested" person. Only few "nerds" will save Idena

**Reply by hlolve** [reply]
We need that "nerds" happy in the group#1 leave the mass in the group#2

**Reply by 0JustinMiles0** [reply]
I'm against trading-off sybil resistance for more users. The [virtual identity staking](https://github.com/idena-network/idena-docs/discussions/178) proposal seems to present better trade-off to grow our user base without sacrificing sybil resistance of the network.

**Reply by hlolve** [reply]
Is not to bring more users, is to adapt to reality, something Idena never did, and maybe that helps to users not lo leave the network. You will not have any sybil resistance to an AI attack with bad quality flips and reports.

**Reply by 0JustinMiles0** [reply]
I already addressed your comment on flips quality in previous comments so unless new arguments are brought up, I'll stop here.

**Comment by simontakouridis** [comment]
Firstly, I love the concept of having an isolated group 2 of identities that only participate in solving flips. They piggy back on the consensus reached in group 1 and therefore do not influence the sybil-resistant properties of the network. HOWEVER, this proposal in its current form does not adequately separate and isolate group 2 and therefore reduces the network security. The following changes are recommended:

1. Identities should not be able to switch between group 1 and group 2, and visa versa. When joining Idena, a user must select whether they are in group 1 or group 2, and this will not change for the life of that identity.
2. Group 2 identities should only have the ability to invite group 2 identities. However, Group 1 identities can invite in either group 1 or Group 2.
3. The size of Group 2 in terms of numbers should be limited by proportionality to Group 1, e.g. If group 1 has 1,000 identities, then group 2 must not exceed 10,000 identities (by disallowing new group 2 members from joining the network). This is not about sybil resistance of the Idena Network, but instead maintaining integrity in the identity solution of Idena for 3rd parties e.g. Gitcoin Grants, where a designated "Idena human" is very likely to be an actual human.

If this proposal is altered as suggested, we could see mass adoption all whilst maintaining security. I look forward to see how this evolves.

**Reply by hlolve** [reply]
Group#2 can´t in any way affect network security, their mining capability is disabled. IDs in group#2 are ruled by IDs in group#1. Invites will not affect, but I considered increasing the possibility to group#1 receive majority of invites. Any ID invited (candidate) will go to group#2 and only if that ID wants to stake the minimum required will be able to select submitting flips during next epoch. Anyway I don´t see any problem to limit at all the capability for group#2 to invite.
I don´t see the problem of IDs switching groups for next epoch. An ID in group#2 can learn an decide to be part of group#1
Is important the proportion of IDs between groups, but is more important to reach a minimum number of IDs in group#1, and that number must be determined.

**Reply by simontakouridis** [reply]
My concern is that Group 2 has less sybil resistance than Group 1 - why? Because in aggregate, the same number of flips will be challenged to both groups, yet Group 2 will have many more identities and therefore a greater sharing of any given individual flip. As such, we cannot allow an identity to reach a given level of trust in the network through Group 2 and then switch over to Group 1.

Also, having a limit on the size of Group 2 in relation to Group 1 is also important so that sybil resistance, within Group 2 only, does not break down and 3rd party apps e.g. Gitcoin Grants, consider an Idena Human to be meaningless.

**Reply by hlolve** [reply]
If an ID from group#2 then goes to group#1, it will have the same job as any ID from group#1 (same as currently).
Group#2 might be less resistant to sybil only if we don't have the minimum number of flips in group#1 or a percentage ratio between 2 groups.
With validation sharding it is said that a group of 5000 validators is as secure as any other group of other 5000 IDs, we need to set that secure point between the minimum participants in group#1 and the top number in group#2 and considering the fragmentation model.

**Reply by hlolve** [reply]
Of course if we limit the capacity of group#2 we need to have a way for that, could be possible to suspend IDs of low score or low age to participate at lottery time. In that way is similar to current mode, and similar to group#1 if the ID forget to submit flips is not allowed to participate in ceremony.

**Reply by simontakouridis** [reply]
I don't think we need a hard limit on the capacity of group#2. Instead we could have a soft limit whereby if the capacity is reached (i.e. the proportional size of group#2 to group#1), no further identities are allowed to join group#2. This is much better for the user experience. Of course, the soft limit could be exceeded if the number of identities in group#1 shrinks. However, if we are conservative with the soft limit that is set, this shouldn't be an issue.

Or we could have both a soft-limit and a hard-limit for resiliency.

**Comment by ubiubi18** [comment]
> Please review and discuss about this
> 
> https://github.com/hlolve/non-mandatoryflips/blob/main/README.md

I like the proposal and would love to see it happen. In the last 18 month i had my eyes less on the identity count but more on the mining node count. The later one is more or less stable, going a few dozens up and down. Rather down. I think those identities that run mining nodes should be more destinct from IDs that dont care about mining. But the community made a good point regarding the switch from group 2 to group 1. Maybe that should be reconsidered. What about a completely distinct identity status for mining humans, that requires identities to go to the whole process of newbie-> verified -> human again, if nonmandative flip identities want to switch to the group eligable for mining?

**Reply by hlolve** [reply]
This is a draft proposal and of course we can make that if an ID decided once not to submit flip, the only way to go to group#1 is start from 0 age.
Also as Simon mentioned is possible to make that only IDs in group#1 can invite.
I am not worried by group#2 if we have the minimum in group#1.
But lets think about a possible group#2 attack.
The only attack they could try is a suddenly switch of many IDs to group#1 and submitting in that epoch all shit flips (1-2-3-4) and in high quantity auto approve by not reporting them.
This scenario could happen also right now in hands of a big farm pool or collusion of them.
But I see this as a really weird scenario.

**Reply by 0JustinMiles0** [reply]
I don't think this is the only attack that this change would enable. If a farmer is able to validate more identities with the same number of workers after the change, it would be an attack on the sybil resistance properties of the protocol. Maybe it wouldn't put at risk the integrity of the L1 chain but it would degrade the properties of Idena as a identity protocol. In theory, I like the idea that we can facilitate people from validating their identity but we have to be very clear about the tradeoffs. 

We need to clearly understand the tradeoff such as we can say:
Today a farm can validate 3 Human identities per worker VS post implementation, a farm will be able to validate n Human identities per worker. 

Intuitively, I would think that we probably can't allow more than the same amount of validated identities in Group#2 that there are in Group#1 and even with this parameter the sybil resistance properties of Idena will be degraded. 

It also brings the question of how do we invite identities in Group#2 vs Group#1. Since there is a limited number of seats, joining Group#2 can't be open to anyone without filter. There will need to be some sort of filtering system such as the invite system that can't be gamed by bots. 

Again, although I like the idea that we can facilitate people from validating their identity, I'm skeptical that we can do so in a way that can significantly increase the number of overall participants (group#1 + group#2) without drastically degrading the sybil resistance properties of the protocol. 

I certainly encourage more research to be done to better understand the tradeoffs of this proposal on how various parameters of this system would impact the sybil resistance properties of the protocol.

**Reply by 0JustinMiles0** [reply]
One option that could enable more scalability for this system would be to allow some flip makers to make more flips than the restricted 5 per identity but this would require a very solid flip vetting process (better than the current reporting system) ideally based on the AI-resistance of the flips. Ultimately the number of identities allowed to participate in Group#2 should be bounded by the number of flips submitted. Unfortunately, I don't think we can take any shortcut on this constraint without giving up on Idena as a trustable anti-sybil solution.

**Reply by hlolve** [reply]
> 
> Today a farm can validate 3 Human identities per worker VS post implementation, a farm will be able to validate n Human identities per worker.
> 

That will change only if there are a really small quantity of flips and a farm pool control a high quantity of candidates.

But with current rules there are 3 possibles attacks from farm pools controlling many IDs that can be done. One is submit intentionally bad flips 1-2-3-4 The other 2 I will not say because they are not obvious and will be mitigated with these changes, we need that farm pools submit less quantity of flips.

The network will be safer if farm pools submit less centralized flips.

**Reply by hlolve** [reply]
> 
> I certainly encourage more research to be done to better understand the tradeoffs of this proposal on how various parameters of this system would impact the sybil resistance properties of the protocol.
>

Nothing is worst against sybil resistance than centralized flip creation, and all farm pools submit flips created for only a few workers.

**Reply by 0JustinMiles0** [reply]
> That will change only if there are a really small quantity of flips and a farm pool control a high quantity of candidates.

I agree that number of flips and number of identities controlled by a farm pool are the two variables that will affect the sybil resistance properties under this new system but I don't think we can say that anti-sybil resistance properties will only be affected in extreme conditions (low number of flips and high number of identities controlled by farm pool). A simulation needs to be conducted to understand the exact impact. 

> we need that farm pools submit less quantity of flips

Maybe you're not implying the following but how would this new system incentivize farm pools from submitting less flips?

Also, am I understanding correctly that under this new system, identities from group#2 would earn no reward? In this case, how would they be able pay for remote nodes access without purchasing iDNAs? 

Also another impact to better understand from this new system is the impact on UX and onboarding to Idena. Idena is already complicated enough to understand and this system would require a lot of thoughts in revamping the UX to make it understandable for newcomers (joining group#1 vs joining group#2, how is the onboarding different, which category fits which user ect.).

**Reply by hlolve** [reply]
> 
> Maybe you're not implying the following but how would this new system incentivize farm pools from submitting less flips?
> 

Farm pools don´t stake, and IDs will need stake to submit flips, same as vote discrimination

> Also, am I understanding correctly that under this new system, identities from group#2 would earn no reward? In this case, how would they be able pay for remote nodes access without purchasing iDNAs?

Usually IDs that came from gitcoin are not interested in rewards, but they still can earn candidate rewards, and stake rewards, and invite rewards if we allow invites from group#2

**Reply by hlolve** [reply]
> 
> Also another impact to better understand from this new system is the impact on UX and onboarding to Idena. Idena is already complicated enough to understand and this system would require a lot of thoughts in revamping the UX to make it understandable for newcomers (joining group#1 vs joining group#2, how is the onboarding different, which category fits which user ect.).
>
For candidates only will be required to accept or not participate in submitting flips.

**Reply by 0JustinMiles0** [reply]
> Farm pools don´t stake, and IDs will need stake to submit flips, same as vote discrimination

I see what you mean but considering that a large share of the rewards farm pools earn is from making flips, it's likely that farm pools will stake the minimum required in order to submit flips and earn flips rewards. 

Also this makes me think of another potential issue this system could introduce with new participants who don't have enough stake to not be discriminated. It seems we could easily reach a group#2 to group#1 ratio that will cap the acceptance of new participants. Ultimately, the growth of the network will be restricted by the number of participants willing to buy the minimum volume of iDNAs required to join group#1 (this will introduce significative friction). The main goal of this proposal being to allow more participants to join, this needs to be carefully considered as it may create the inverse of the intended effect.

**Reply by 0JustinMiles0** [reply]
> > Also another impact to better understand from this new system is the impact on UX and onboarding to Idena. Idena is already complicated enough to understand and this system would require a lot of thoughts in revamping the UX to make it understandable for newcomers (joining group#1 vs joining group#2, how is the onboarding different, which category fits which user ect.).
> 
> For candidates only will be required to accept or not participate in submitting flips.

So how would new candidates join the network when we reach the group#2 to group#1 max ratio? Also, I'm concerned that this system relies on the number of minimum stake to avoid discrimination as this concept is not understood at all by new participants. There isn't even any information in the UI about how identities have their vote not being counted and what they could do for it to count. This would require to be clearly communicated in the UI.

**Reply by hlolve** [reply]
Currently validators that don´t have big stake receives almost nothing for flips, invites or mining rewards, and are forced to submit flips if want to keep an ID

**Reply by 0JustinMiles0** [reply]
> Currently validators that don´t have big stake receives almost nothing for flips, invites or mining rewards, and are forced to submit flips if want to keep an ID

But apparently farm pools receive enough rewards to keep them in business and flip rewards is a large share of their rewards. They will definitely stake the minimum required to keep earning flip rewards. So I don't think we can assume that this change would reduce farm pools participating in flip making. More concerning, by incentivizing farm pools to reach the minimum stake it gives them more voting power which will reintroduce the attack vector that minimum stake discrimination aims to suppress.

**Reply by hlolve** [reply]
The main problem is that new candidates are forced to learn a lot of rules. For they is easy to pass the flips test but not reporting or worse submit flip that later will be reported. IDs that only wants to prove humanness don´t need to be forced for all of this

**Reply by hlolve** [reply]
> 
> But apparently farm pools receive enough rewards to keep them in business and flip rewards is a large share of their rewards. They will definitely stake the minimum required to keep earning flip rewards (which by the way will give them more voting power). So I don't think we can assume that this change would reduce farm pools participating in flip making.
>
Yes, because they submit flips that are not reported. I am not against pools if they submit good flips

**Reply by 0JustinMiles0** [reply]
> The main problem is that new candidates are forced to learn a lot of rules. For they is easy to pass the flips test but not reporting or worse submit flip that later will be reported. IDs that only wants to prove humanness don´t need to be forced for all of this

In this case, maybe a simpler solution is to simplify the rules. I've proposed a simplified framework but I'm sure there are other that could be implemented:

- Approve flips that are highly or moderately interesingly unique
- Abstain for flips that are slightly interestingly unique
- Report flips that are not interestingly unique (even if they contain the two required keywords)
- Report flips that don't contain the two required keywords regardless of how interestingly unique they are
- Report flips with offensive/NSFW content regardless of how interestingly unique they are

**Reply by 0JustinMiles0** [reply]
> > But apparently farm pools receive enough rewards to keep them in business and flip rewards is a large share of their rewards. They will definitely stake the minimum required to keep earning flip rewards (which by the way will give them more voting power). So I don't think we can assume that this change would reduce farm pools participating in flip making.
> 
> Yes, because they submit flips that are not reported. I am not against pools if they submit good flips

But again, I don't see how this system will disincentivize farm pools from submitting flips. All, they need to do is to stake to the minimum required amount.

**Reply by hlolve** [reply]
I am perfect with current rules, buy I am a "nerd" and maybe there are other 300 like me that are happy with these rules, but if we don't see the reality of people leaving the network or the improvement in AI, we will not be happy in a near future

**Reply by 0JustinMiles0** [reply]
It's not that I'm not seeing the current problems but that I'm not convinced that this proposal will fix them without introducing other similar problems.

**Comment by ubiubi18** [comment]
@midenaio is it fruitful if the community still discusses IIPs, or do you have no capacity currently and we should wait for better times?

**Reply by sekisanchi** [reply]
@ubiubi18 copy to @midenaio Hi, These are a bit behind from a "design proposal", which leads into a "design change"  We do have both user and designer roles here, but not clear where in the protocol have to be changed within the current protocol model consistency.
On the other hand, in theory, any person or party can "fork" to start new idena protocol as seen in bitcoin and ethereum. 
Considering the scale of changes from small work around fix to hard fork, impact of this requirement is sitting at midium or midium-high, 
and thus it requires a bit specific as an IIP which includes both protocol change detail and the impact to the current model. 
I have not seen that within above chronologies yet.

**Comment by morienzo** [comment]
Thank you, Hlove, for starting the discussion. Here are a few elements I would like to propose:

- The number of flips that one can create is determined by age or stake:

Age below 25 or stake below 25K IDNA: 5 Flips
Age between 25 and 50 or stake between 25K and 50K IDNA: 7 Flips
Age between 50 and 75 or stake between 50K and 75K IDNA: 9 Flips
...

- 1 reported flip results in rewards being halved. 2 reported flips result in 0 rewards.

- To achieve human rank, one must be a flip creator (and not have 2 reported flips).

- All addresses with 2 reported flips or that do not create flips will have a "Verified" rank.

- Only "Verified" or "Human" addresses can create flips.

- Only Humans can earn mining rewards during the next epoch.

- A minimum of 3 flips is required for all creators.

- All flips beyond 3 receive quadratic rewards.

- Keywords are distributed dynamically: 9 pairs of keywords initially, and then a new pair is received for each flip created (if a flip is deleted, no new pair is received).

**Reply by hlolve** [reply]
Yes, we need a way to allow an ID submit more flips.
Here is also an alternative.
https://github.com/hlolve/Flip-packs/blob/main/README.md

**Reply by morienzo** [reply]
I am rather against the 60k threshold idea because it would increase the risk of abuse. I think the current distribution system works quite well, so I would keep it as it is (stake^0.9 for each extra flip 4 & 5 per pack).

Here's what I would prefer for access to flip packs:

1x Vote discrimination (~600 iDNA) = 1 flip pack
5x Vote discrimination (~3000 iDNA) = 2 flip packs
10x Vote discrimination (~6000 iDNA) = 3 flip packs
15x Vote discrimination (~9000 iDNA) = 4 flip packs
20x Vote discrimination (~12000 iDNA) = 5 flip packs

And I would limit the creation of flips to Humans only.

**Reply by hlolve** [reply]
The top 60k threshold idea is to avoid the risk of abuse by high stakers and allow more incentives for lower stakers.
With packs the risk to submit extra flips will be split in each pack.
I like the idea to allow packs by stake and not by type of identity, but I will allow to any validated ID to submit flips.
The restriction will be by stake, and newbie and verified don´t have stake protected.

**Reply by morienzo** [reply]
True, the lack of stake protection is a hindrance in itself, so the minimum rank is not necessary (everyone is free to take the risk). Regarding the 60K limit, I find the ambition to share rewards fairly is a good thing, but the method bothers me. It is logical for a quadratic reward that the person with a high stake is better rewarded. These quadratic rewards only concern a portion of the flips that will be made and should not be limited. However, I would support opening a discussion on the distribution of rewards in case of network modification, but I think that should be done later, through another IIP. There are several solutions to consider for a fairer distribution:

- Modify the number of extra flips possible per pack
- Modify the parameter K of the stake power
- Modify the percentages attributed to each type of flip

Passing this IIP without modifying the distribution would also allow us to gather statistics on the rewards. Today, all of this is hypothetical.

**Reply by hlolve** [reply]
The upper stake limit can be removed. We can see first how "flip packs" works and after modify rewards incentives if required.

**Reply by sekisanchi** [reply]
Sorry to cut into the middle.  I agree with the direction of number of flips with staking incentives, but have a large concern on heavily biased flips dominance. With current flip slashing rules and frequent late submission errors under the noise induced FLIPs, it will increase first onboarding hurdle, and thus dis-incetivize independent or non organized brand new FLIP creators to participate.         
I want some plan to improve heavily biased, complex and ugly FLIPS as a pre-req.  
Attached below is the best FLIP I've ever seen, but this type of FLIPs or creator will may be defeated by strange biassed mass production.  In another word, I want to prioritize the fundamental health surgery than incentives.   
Snake/Cane  https://scan.idena.io/flip/bafkreifxwsimn5rels5thqmmht6qubplzvbbgx5gv77ldsqk2tfpmr56be
![SnakeCane01](https://github.com/idena-network/idena-docs/assets/1669550/07084293-d5f2-42bd-a6a3-dd66509b2d9a)

**Reply by hlolve** [reply]
Late submission error is a problem from users that don´t know how to participate in the ceremony.
I and my family/friends never had a late submission, using desktop app or web app in more than 80 ceremonies.
If we allow people that only want to prove is "human" and not interested in Idena, we are exposed to bad flips.
Flips production and reporting is based in "rewards", if someone participating don't care of that rewards you need to isolate that people, but is possible to give them their "human" status only solving flips.
That is the purpose if this proposal.
